### PR TITLE
Add PHP GitLab CI setup documentation

### PR DESCRIPTION
**Pull request by LakshayaT**
Task ID: REST-64

This PR adds the Gitlab CI setup documentation for PHP projects.

### CODE REVIEW

File: ci-setups/php-gitlab.md
Comment (osm-vishnukyatannawar) [inline ci-setups/php-gitlab.md:44]: give sample with multiple target branches.

File: ci-setups/php-gitlab.md
Comment (osm-vishnukyatannawar) [inline ci-setups/php-gitlab.md:84]: I want to allow multiple branches.

I mean generally people will have more than 1 main target branch.

ex: main, release, development etc

So can you update this so that it can take multiple branches and then we let user decide what they want to configure as per their project need.

Ensure that you test it in sample project.

Try this since this is easier and looks clean, the || (or condition) looks big . But answer from chatgpt so cannot trust.

```sh
# Define the rules for pipeline triggering
# This will trigger the pipeline for merge requests targeting main, dev, or staging branches
# Adjust the branches and rules as needed
rules:
  - if: '$CI_PIPELINE_SOURCE == "merge_request_event"'
    changes:
      - main
      - dev
      - staging
```

File: ci-setups/php-gitlab.md
Comment (osm-vishnukyatannawar) [inline ci-setups/php-gitlab.md:102]: I would not do this.

If we want to update the dependencies then it has to be updated in local and tested first.

But if in local I don't update and also on server where I deploy code I don't update then no point of updating and checking in CI.

Better to remove.

File: ci-setups/php-gitlab.md
Comment (osm-vishnukyatannawar) [inline ci-setups/php-gitlab.md:140]: Can we also add 1 section to ask the maintainer to make the following changes.

1. PR check should pass before merging.
2. If CI check doe snot pass then we should not allow to merge
3. But should let some leads to bypass it (because there can be some cases where we want that to happen)

**Review by osm-vishnukyatannawar — CHANGES_REQUESTED** [review]

File: ci-setups/php-gitlab.md
Comment (LakshayaT) [reply ci-setups/php-gitlab.md]: This has been addressed in the latest commit.

File: ci-setups/php-gitlab.md
Comment (LakshayaT) [reply ci-setups/php-gitlab.md]: This has been addressed in the latest commit.
New variables have been defined with regex pattern for matching branch names for push and merge requests.

File: ci-setups/php-gitlab.md
Comment (LakshayaT) [reply ci-setups/php-gitlab.md]: This has been removed in the latest commit along with the step creation of .env file.

File: ci-setups/php-gitlab.md
Comment (LakshayaT) [reply ci-setups/php-gitlab.md]: This section has been added in the documentation restructuring

**Review by osm-vishnukyatannawar — APPROVED** [review]

**Pull request merged**
